megamek/src/megamek/common/Compute.java

```diff
@@ -5457,6 +5457,34 @@ public static int dialDownHeat(Mounted weapon, WeaponType wtype, int range) {
         return entities;
     }
 
+    public static boolean isInUrbanEnvironment(IGame game, Coords unitPOS) {
```

Comment:
You end up checking the unit hex 6 times.  This seems superfluous.  I would like to see the unitHex pulled outside of the loop, so that the loop only checkes the `hextor` IHex (I also feel like `hextor`variable name could be better; it doesn't seem to mean anything to me, I would've gone with `adjHex` or something similar).  Likewise, `tempcoords` could be `adjCoords`.